lib/elixir/lib/file/stream.ex

```diff
@@ -88,12 +90,46 @@ defmodule File.Stream do
       Stream.resource(start_fun, next_fun, &:file.close/1).(acc, fun)
     end
 
-    def count(_stream) do
-      {:error, __MODULE__}
+    def count(%{path: path, modes: modes, line_or_bytes: :line} = stream) do
+      modes = modes -- [:write, :append]
+      pattern = :binary.compile_pattern("\n")
+      counter = &count_lines(&1, path, pattern, read_function(stream), 0)
+
+      case File.open(path, modes, counter) do
+        {:ok, count} ->
+          {:ok, count}
+        {:error, reason} ->
+          raise File.Error, reason: reason, action: "stream", path: path
+      end
+    end
+
+    def count(%{path: path, line_or_bytes: bytes}) do
+      case File.stat(path) do
```

Comment:
I think we should but if we lose the trivial O(1) count I would be indifferent to this getting in. Are we going to continue to add 30LOC optimisation to outlier functions like that make the code less clear?